src/main.cpp

```diff
@@ -18,48 +21,34 @@ static std::vector<std::vector<std::string>> levels;
 
 extern "C" {
 void sokoban_initialize() {
-    levels = {
-        {
-            // greg messing around
-            "   ####",
-            "####. ##",
-            "#.##$ .##",
-            "# ##    #",
-            "#    $  #",
-            "# #### ##",
-            "#@ $   #",
-            "#####  #",
-            "    ####",
-        },
-        {
-            // GRIGoRusha: Shito-Krito #86
-            "  #########",
-            "  #   #  @#",
-            " ##  .$.# #",
-            " #  ##  $ #",
-            "##  .$*#$##",
-            "#  #$#   # ",
-            "#  . .# ## ",
-            "####    #  ",
-            "   ######  ",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
+    std::filesystem::path getLevelsDir = std::filesystem::directory_entry("src/assets/levels");
+    std::vector<std::filesystem::path> get_map_path; // save path into vector element
+
+    for (const auto& entry : std::filesystem::directory_iterator(getLevelsDir)) {
+        get_map_path.push_back(entry.path());
```

Comment:
I'm still unclear why we need a vector here. We're reading everything into the vector using a loop, then looping over the vector again, then throwing the vector away. Can't we use this loop for all of the logic directly?

Reply:
I think this vector is for containing all the filenames/filepaths read from a directory.

Reply:
Yes. I'm asking why we can't do all of that in one loop? Open the file as soon as you have the name, one loop!